rpc/connection.go

```diff
@@ -615,6 +617,17 @@ func (c *Connection) connect(ctx context.Context) error {
 		return err
 	}
 
+	// If we encounter any errors before successfully completing the connection,
+	// close the transport to prevent resource leaks and duplicate servers.
+	// This provides immediate cleanup without waiting for Shutdown() to be called.
+	defer func() {
+		if err != nil {
+			c.log.Debug("Connection: %s",
+				LogField{Key: ConnectionLogMsgKey, Value: "closing transport due to connection failure"})
+			transport.Close()
```

Comment:
https://github.com/keybase/go-framed-msgpack-rpc/pull/121/ would perform this close on any retry to `Dial` but that leaves whatever the retry interval is/until `Shutdown` is called for the connection to remain open and possible receive data